xdsl/dialects/ptr.py

```diff
@@ -0,0 +1,70 @@
+from xdsl.dialects.builtin import IndexType, IntegerType, UnitAttr
+from xdsl.ir import Dialect, ParametrizedAttribute, TypeAttribute
+from xdsl.irdl import (
+    AnyOf,
+    IRDLOperation,
+    irdl_attr_definition,
+    irdl_op_definition,
+    operand_def,
+    opt_prop_def,
+    result_def,
+)
+
+
+@irdl_attr_definition
+class PtrType(ParametrizedAttribute, TypeAttribute):
+    name = "ptr.ptr"
+
+
+@irdl_op_definition
+class PtrAddOp(IRDLOperation):
+    name = "ptr.ptradd"
+
+    addr = operand_def(PtrType)
+    offset = operand_def(AnyOf([IntegerType, IndexType]))
+    result = result_def(PtrType)
+
+    assembly_format = "$addr `,` $offset attr-dict `:` `(` type($addr) `,` type($offset) `)` `->` type($result)"
+
+
+# haven't managed to pass a type here yet. so did it with a hack.
```

Comment:
The way that should work but is a bit inelegant is `base(IntegerType) | base(IndexType)`, there might be `IntegerTypeConstr` already defined, if not `BaseAttr(IntegerType)` should also work

Reply:
I got it to work but only via passing the type through attributes. Is it possible to pass it as an operand? Xdsl complains that it expects an operand there if write `"op"(i32)`